fedimint-client/src/transaction/mod.rs

```diff
@@ -1,5 +1,114 @@
 mod builder;
 mod sm;
 
+use std::any::Any;
+use std::fmt::{Debug, Display, Formatter, Pointer, Write};
+use std::io::Error;
+
 pub use builder::*;
+use fedimint_core::core::{IntoDynInstance, ModuleInstanceId};
+use fedimint_core::encoding::{Decodable, DynEncodable, Encodable};
+use fedimint_core::secp256k1::schnorr::Signature;
+use fedimint_core::secp256k1::{KeyPair, Message, Secp256k1};
+use fedimint_core::task::{MaybeSend, MaybeSync};
+use secp256k1_zkp::All;
 pub use sm::*;
+
+pub trait SchnorrSigner: Clone + MaybeSend + MaybeSync {
```

Comment:
In case anyone ever picks it up: probably better to have a one "`DynSigner`" that is resonsible for all client crypto operations.